pkg/expapi/types.go

```diff
@@ -27,3 +27,40 @@ support is experimental.
 */
 
 package expapi
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// SubresourceReference contains enough information to let you inspect or modify the referred subresource.
+type SubresourceReference struct {
+	Kind        string `json:"kind,omitempty"`
+	Namespace   string `json:"namespace,omitempty"`
+	Name        string `json:"name,omitempty"`
+	APIVersion  string `json:"apiVersion,omitempty"`
+	Subresource string `json:"subresource,omitempty"`
+}
+
+// HorizontalPodAutoscaler represents the configuration of a horizontal pod autoscaler.
+type HorizontalPodAutoscaler struct {
+	api.TypeMeta   `json:",inline"`
+	api.ObjectMeta `json:"metadata,omitempty"`
+
+	// Target is a reference to Scale subresource. HorizontalPodAutoscaler will learn the current resource consumption from its status,
+	// and will set the desired number of pods by modyfying its spec.
+	Target *SubresourceReference `json:"target"`
+	// MinCount is the lower limit for the desired number of pods that can be set by the autoscaler.
+	MinCount int `json:"minCount"`
+	// MaxCount is the upper limit for the desired number of pods that can be set by the autoscaler. It cannot be smaller than MinCount.
+	MaxCount int `json:"maxCount"`
+	// TargetCPUConsumption is the target average CPU consumption of pods that the autoscaler will try to maintain by adjusting the desired number of pods.
+	TargetCPUConsumption float64 `json:"targetCPUConsumption"`
```

Comment:
I assume the desire here was to focus on a concrete use case that can be achieved.  Is the concern with the previous intents proposal is that it was too broad and that you don't believe the full flexibility is needed?  Or only that having a practical example is important in the short term?

One of our guiding principles is that we know customers to want to autoscaler based on more than CPU.  We had assumed the previous proposal would give us some of that flexibility.  Is there a reason not to carry some of that flexibility over?